Fix Travis log parsing error introduced by #380.

heplesser opened the pull request:
After merging #380, the Travis log was no longer parsed correctly if clang-format led to diffs. This is fixed here.

heplesser commented:
@tammoippen @apeyser Could you review this one ASAP, together with #380 that should get Travis moving again.

Once it is merged we should alert every owner of open PRs to merge from master to make sure their PRs build on Travis again.

apeyser commented:
@heplesser 

What happened with the Travis log files? It's not obvious to me what this is supposed to be fixing.

heplesser commented:
@apeyser #380 adapted our Travis setup after the APT repository for clang was no longer available. Unfortunately, that change introduced some subtle changes (e.g., that the clang formatter executable is now `clang-format` instead of `clang-format-3.6`), which meant that `parse_travis_log.py` could no longer parse the build log correctly. Parsing failed in particular if errors had been detected, e.g., by the clang formatter or PEP8. This PR attempts to adapt the parsing script to the changes in the log file.

In addition (and that was maybe not the brightest choice, since it mixes two things), the PR also suppresses sections of the final summary report that are not relevant in a given case (e.g., the static analysis section if static analysis did not turn up any problems).

 [Travis Build 840](https://travis-ci.org/nest/nest-simulator/builds/136129882) which includes the fixes here per (f5dc375) shows that logs with ill-formatted C++ code are correctly parsed now, i.e., reported as failure with a proper final report.

tammoippen commented:
👍  from my side.

sanjayankur31 commented on build.sh:
```diff
@@ -100,11 +100,13 @@ if [ ! -f "$HOME/.cache/bin/cppcheck" ]; then
+  rm -rf ./clang+llvm-3.6.2-x86_64-linux-gnu-ubuntu-14.04
 fi
 
 export PATH=$PATH:$HOME/.cache/bin
```
I still think the new path should be added to the head of PATH, not the tail:

```bash
export PATH=$HOME/.cache/bin:$PATH
```

This is what the build output from travis says:
```bash
echo $PATH

/home/travis/bin:/home/travis/.local/bin:/home/travis/.rvm/gems/ruby-2.2.3/bin:/home/travis/.rvm/gems/ruby-2.2.3@global/bin:/home/travis/.rvm/rubies/ruby-2.2.3/bin:/home/travis/.pyenv/shims:/home/travis/.phpenv/shims:/home/travis/.php-build/bin:/home/travis/.local/bin:/home/travis/.pyenv/shims:/opt/pyenv/bin:/home/travis/.phpenv/shims:/home/travis/perl5/perlbrew/bin:./node_modules/.bin:/usr/local/clang-3.5.0/bin:/home/travis/.local/bin:/home/travis/.pyenv/shims:/opt/pyenv/bin:/home/travis/.phpenv/shims:./node_modules/.bin:/usr/local/clang-3.5.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/travis/.phpenv/bin:/home/travis/.phpenv/bin:/home/travis/.rvm/bin:/home/travis/.rvm/bin
```

It has clang-3.5 there already in the path, this version will be invoked if we append to the tail of PATH, right?

sanjayankur31 replied:
Ah, you fixed it by the time I posted my comment. :D

heplesser commented:
Testing a test system is a bit cumbersome, but I think I have all the bits in place now. I have now tested that proper reports are created if either Vera, clang-format og pep8 report problems, as well as if nothing goes wrong. To test reports on pep8 errors, I introduced an problem intentionally in 0e8afb2 (Travis run 850) and removed it again with f0db84f (Travis run 851). The respective test cases for C++ problems and testsuite failures are Travis runs for #387 (846 clang-format reported diff, 848 test suite found error, 849 all fine).

@apeyser Would you take a look before we merge?

apeyser commented:
:+1:
